crates/storage/src/entities.rs

```diff
@@ -644,17 +644,19 @@ impl Element {
     /// # Parameters
     ///
     /// * `path` - The path to the [`Element`] in the hierarchy of the storage.
+    /// * `id` - The id of the [`Element`] in the hierarchy of the storage.
     ///
     /// # Panics
     ///
     /// This method can technically panic if the system time goes backwards, to
     /// before the Unix epoch, which should never ever happen!
     ///
     #[must_use]
-    pub fn new(path: &Path) -> Self {
+    pub fn new(path: &Path, id: Option<Id>) -> Self {
```

Comment:
this should always be random, save for the root, which already has it's own function below `Element::root()`